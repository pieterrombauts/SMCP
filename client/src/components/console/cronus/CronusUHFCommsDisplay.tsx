import React from 'react';
import styled from 'styled-components';

interface CronusUCDProps {
  className?: string;
};

const UCronusUHFCommsDisplay: React.FC<CronusUCDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_2.png'></image>
        
        <rect x="50" y="405" rx="5" ry="5" width="510" height="200" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
        
        <text x="6%" y="45%" fill="white" font-size="25" font-family="Verdana">Space to Space Station</text>

        <text x="8%" y="48%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
        <text x="25%" y="48%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="8%" y="51%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
        <text x="25%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="6%" y="57%" fill="white" font-size="25" font-family="Verdana">UHF Sync Lcok</text>
        <text x="28%" y="57%" fill="#d6ab62" font-size="20" font-family="Verdana">Frame Sync</text>
        
        <rect x="815" y="435" rx="5" ry="5" width="410" height="140" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
        
        <text x="65%" y="49%" fill="white" font-size="20" font-family="Verdana">IAC1 Power</text>
        <text x="78%" y="49%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

        <text x="65%" y="53%" fill="white" font-size="20" font-family="Verdana">IAC2 Power</text>
        <text x="78%" y="53%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>
      </svg>
    </div>
  )
};

const CronusUHFCommsDisplay = styled(UCronusUHFCommsDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusUHFCommsDisplay;