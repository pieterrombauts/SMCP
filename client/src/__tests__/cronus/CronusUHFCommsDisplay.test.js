import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import CronusUHFCommsDisplay from '../../components/console/cronus/CronusUHFCommsDisplay'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  sBand: {
    elevationGimbal: 60,
    azimuthGimbal: 2
  }
}

describe("Cronus UHF communications display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<CronusUHFCommsDisplay values={defaultValues} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})