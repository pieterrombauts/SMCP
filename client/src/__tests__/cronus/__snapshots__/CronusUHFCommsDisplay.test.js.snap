// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus UHF communications display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UCronusUHFCommsDisplay)
  values={
    Object {
      "sBand": Object {
        "azimuthGimbal": 2,
        "elevationGimbal": 60,
      },
    }
  }
>
  <UCronusUHFCommsDisplay
    className="c0"
    values={
      Object {
        "sBand": Object {
          "azimuthGimbal": 2,
          "elevationGimbal": 60,
        },
      }
    }
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Cronus_2.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="200"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#36485b",
              "stroke": "#2a3a3f",
              "strokeWidth": "5",
            }
          }
          width="510"
          x="50"
          y="405"
        />
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="25"
          x="6%"
          y="45%"
        >
          Space to Space Station
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="8%"
          y="48%"
        >
          Radio1 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="25%"
          y="48%"
        >
          On
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="8%"
          y="51%"
        >
          Radio1 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="25%"
          y="51%"
        >
          On
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="25"
          x="6%"
          y="57%"
        >
          UHF Sync Lcok
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="28%"
          y="57%"
        >
          Frame Sync
        </text>
        <rect
          height="140"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#36485b",
              "stroke": "#2a3a3f",
              "strokeWidth": "5",
            }
          }
          width="410"
          x="815"
          y="435"
        />
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="65%"
          y="49%"
        >
          IAC1 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="78%"
          y="49%"
        >
          Active
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="65%"
          y="53%"
        >
          IAC2 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="78%"
          y="53%"
        >
          Active
        </text>
      </svg>
    </div>
  </UCronusUHFCommsDisplay>
</Styled(UCronusUHFCommsDisplay)>
`;
